### PR TITLE
Fixes #32410 - use DSL to define settings

### PR DESCRIPTION
**Pull request by ezr-ondrej**

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
@adamruzicka care to take a look, or someone from your team?

**Comment by adamruzicka** [comment]
Can't assign myself (or anyone really) as a reviewer, but I'll take it for a spin later today

**Review by adamruzicka — COMMENTED** [review]
One nitpick, otherwise lgtm

File: lib/foreman_templates.rb [hunk 1]
```diff
@@ -4,4 +4,16 @@ module ForemanTemplates
   BASE_SETTING_NAMES = %w(repo branch dirname filter negate).freeze
   IMPORT_SETTING_NAMES = (BASE_SETTING_NAMES | %w(prefix associate force lock)).freeze
   EXPORT_SETTING_NAMES = (BASE_SETTING_NAMES | %w(metadata_export_mode commit_msg)).freeze
+
+  def self.associate_types
```
[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply lib/foreman_templates.rb:8]: Yeah, that is because of translations... we need to have them evaluated per request, not on constant evaluation.

**Review by adamruzicka — APPROVED** [review]
LGTM, but I don't have merge permissions here

File: lib/foreman_templates/engine.rb [hunk 2]
```diff
@@ -28,10 +24,78 @@ class Engine < ::Rails::Engine
...
+              default: '',
+              full_name: N_('Branch'))
+            setting('template_sync_metadata_export_mode',
+              type: :string,
+              description: N_('Default metadata export mode, refresh re-renders metadata, keep will keep existing metadata, remove exports template without metadata'),
+              default: 'refresh',
+              full_name: N_('Metadata export mode'),
+              collection: -> { ForemanTemplates.metadata_export_mode_types })
+            setting('template_sync_force',
+              type: :string,
```
Comment (ofedoren) [inline lib/foreman_templates/engine.rb:81]: Shouldn't it be a boolean?

Comment (ezr-ondrej) [reply lib/foreman_templates/engine.rb]: good catch, thanks :+1:

**Comment by ofedoren** [comment]
[test foreman_templates]

**Comment by ezr-ondrej** [comment]
:green_apple:

**Comment by ofedoren** [comment]
Thanks, @ezr-ondrej, @adamruzicka, merging :)

**Pull request merged**